tests/e2e/controls.py

```diff
@@ -2717,3 +2758,404 @@ def expect_icon(self, value: PatternOrStr, *, timeout: Timeout = None) -> None:
 
     def expect_open(self, is_open: bool, *, timeout: Timeout = None) -> None:
         _expect_class_value(self.loc_body, "show", is_open, timeout=timeout)
+
+    # user sends value of Open: true | false
+    def set(self, open: bool, *, timeout: Timeout = None) -> None:
+        expect_not_to_have_class(self.loc_body, "collapsing", timeout=timeout)
+        if self.loc_body_visible.count() != int(open):
+            self.toggle(timeout=timeout)
+
+    def toggle(self, *, timeout: Timeout = None) -> None:
+        self.loc.wait_for(state="visible", timeout=timeout)
+        self.loc.scroll_into_view_if_needed(timeout=timeout)
+        self.loc_header.click(timeout=timeout)
+
+
+class _OverlayBase(_InputBase):
```

Comment:
Merge in the latest from `main`. Hopefully most of the new code in the rest of the file is not double added. 

... unless there are new changes?